src/test/rustdoc/fn-bound.rs

```diff
@@ -0,0 +1,17 @@
+use std::iter::Peekable;
+
+pub struct Span<F: Fn(&i32)> {
+    inner: Peekable<ConditionalIterator<F>>,
+}
+
+struct ConditionalIterator<F> {
+    f: F,
+}
+
+impl<F: Fn(&i32)> Iterator for ConditionalIterator<F> {
```

Comment:
Could you add a test like this to make sure the output is correct?

```suggestion
// @has 'fn-bound/struct.ConditionalIterator.html' '//h3[@class="code-header"]' 'impl<F: Fn(&i32)> Iterator for ConditionalIterator<F>'
impl<F: Fn(&i32)> Iterator for ConditionalIterator<F> {
```